### PR TITLE
New function to update config files at run time

### PR DESCRIPTION
**Pull request by cleptric**
This finishes the PR #8452, as the initial creator hasn't updated his since March.

### CODE REVIEW

**Comment by dereuromark** [comment]
Not sure if that is a good idea.
After all, those files are usually in git commited, and that would eventually also make deployments fail due to modified files (requires a reset of those then on top).
It is better to manually do something like that on project level, and for non persistent "app_local" files of some sort.

Changing Confiure::write() on the fly is OK to me, but not the files in version control on the system.
This invites to abuse and I am :-1: on that.

**Comment by cleptric** [comment]
@dereuromark You wrote on the 7th of October in the original PR:

> You need to fix up the PR, only small minor adjustments, and this should be good to go.

So this is all I did.

**Comment by dereuromark** [comment]
You are right, and maybe we should get more feedback from the others.
But after thinking about it some more I do have some concerns about this new feature.
I am sorry that I didnt really take a closer look before and just voiced my concern now that late in the process.

@markstory @lorenzo What do you think?

File: tests/TestCase/Core/ConfigureTest.php [hunk 1]
```diff
@@ -548,6 +548,27 @@ public function testDumpPartial()
...
+     * @return void
+     */
+    public function testUpdateDump()
+    {
+        Configure::config('test_Engine', new PhpConfig(TMP));
+        Configure::dump('config_test', 'test_Engine');
+
+        $uniqueKeyName = 'UniqueKeyName';
+        $result = Configure::updateDump('config_test', [$uniqueKeyName . '.testKey' => 'testValue'], 'test_Engine');
+        $this->assertTrue($result > 0);
```
Comment (markstory) [inline tests/TestCase/Core/ConfigureTest.php]: assertGreaterThan() ?

File: tests/TestCase/Core/ConfigureTest.php [hunk 2]
```diff
@@ -548,6 +548,27 @@ public function testDumpPartial()
...
+
+        $fileContents = file_get_contents(TMP . 'config_test.php');
+        $this->assertContains($uniqueKeyName, $fileContents);
```
Comment (markstory) [inline tests/TestCase/Core/ConfigureTest.php]: Should you also check that some of the other keys were dumped?

**Pull request closed without merging**